src/WebJobs.Extensions.DocumentDB/Bindings/DocumentDBEnumerableBuilder.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Client;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DocumentDB
+{
+    internal class DocumentDBEnumerableBuilder<T> : IAsyncConverter<DocumentDBAttribute, IEnumerable<T>> where T : class
+    {
+        private readonly DocumentDBConfiguration _config;
+
+        public DocumentDBEnumerableBuilder(DocumentDBConfiguration config)
+        {
+            _config = config;
+        }
+
+        public async Task<IEnumerable<T>> ConvertAsync(DocumentDBAttribute attribute, CancellationToken cancellationToken)
+        {
+            DocumentDBContext context = _config.CreateContext(attribute);
+
+            Uri collectionUri = UriFactory.CreateDocumentCollectionUri(context.ResolvedAttribute.DatabaseName, context.ResolvedAttribute.CollectionName);
+
+            List<T> finalResults = new List<T>();
+
+            string continuation = null;
+
+            SqlQuerySpec sqlSpec = new SqlQuerySpec
+            {
+                QueryText = context.ResolvedAttribute.SqlQuery,
+                Parameters = context.ResolvedAttribute.SqlQueryParameters ?? new SqlParameterCollection()
+            };
+
+            do
+            {
+                DocumentQueryResponse<T> response = await DocumentDBUtility.RetryAsync(
+                    () => context.Service.ExecuteNextAsync<T>(collectionUri, sqlSpec, continuation),
+                    context.MaxThrottleRetries);
```

Comment:
404s -- which will only happen if the collection or db don't exist -- will surface as a function error. I checked how Table works in this regard, and it matches that behavior.

Yeah, there were 429 issues that I worked through a while ago. Although I may need to revisit as it appears newest DocDB packages will handle 429s internally now (with a few knobs). So I may not even need any of this code.